### PR TITLE
Support to return prometheus metrics with instance label

### PR DESCRIPTION
**Pull request by turboFei**
### Why are the changes needed?

For my use case, the instances are not human readable, so I prefer to return the FQDN.
image

### How was this patch tested?

Integration testing.
```
(base) ➜  dist git:(prometheus_label_2) cat conf/kyuubi-defaults.conf
kyuubi.metrics.prometheus.metrics.instance.enabled=true
kyuubi.zookeeper.embedded.client.port.address=localhost
kyuubi.frontend.bind.host=localhost
```

image

### Was this patch authored or co-authored using generative AI tooling?

No.

### CODE REVIEW

File: kyuubi-metrics/src/main/scala/org/apache/kyuubi/metrics/MetricsConf.scala [hunk 1]
```diff
@@ -80,6 +80,13 @@ object MetricsConf {
     .checkValue(path => path.startsWith("/"), "Context path must start with '/'")
     .createWithDefault("/metrics")
 
+  val METRICS_PROMETHEUS_METRICS_INSTANCE_ENABLED: ConfigEntry[Boolean] =
+    buildConf("kyuubi.metrics.prometheus.metrics.instance.enabled")
```
Comment (pan3793) [inline kyuubi-metrics/src/main/scala/org/apache/kyuubi/metrics/MetricsConf.scala:84]: maybe
```suggestion
    buildConf("kyuubi.metrics.prometheus.labels.instance.enabled")
```
?

Comment (turboFei) [reply kyuubi-metrics/src/main/scala/org/apache/kyuubi/metrics/MetricsConf.scala:84]: Sounds good.

File: kyuubi-metrics/src/main/scala/org/apache/kyuubi/metrics/PrometheusReporterService.scala [hunk 2]
```diff
@@ -100,4 +112,46 @@ class PrometheusReporterService(registry: MetricRegistry)
...
+    val metricsSnapshotWriter = new java.io.StringWriter
+    val contentType = TextFormat.chooseContentType(null)
+    TextFormat.writeFormat(contentType, metricsSnapshotWriter, bridgeRegistry.metricFamilySamples())
+    val labelStr = labelString(labels)
+    metricsSnapshotWriter.toString.split("\n").map { line =>
+      if (line.startsWith("#")) {
+        line
+      } else {
+        val Array(metrics, value) = line.split("\\s+", 2)
+        s"""$metrics${labelStr} $value"""
```
Comment (pan3793) [inline kyuubi-metrics/src/main/scala/org/apache/kyuubi/metrics/PrometheusReporterService.scala:149]: a little bit hacky, but given the feature is disabled by default, let's merge it in first.

**Review by pan3793 — APPROVED** [review]

File: kyuubi-metrics/src/main/scala/org/apache/kyuubi/metrics/PrometheusReporterService.scala (hunk 2 above)
Comment (turboFei) [reply kyuubi-metrics/src/main/scala/org/apache/kyuubi/metrics/PrometheusReporterService.scala]: changed to

```
       line.split("\\s+", 2) match {
          case Array(metrics, rest) => s"""$metrics${labelStr} $rest"""
          case _ => line
        }
```

**Comment by pan3793** [comment]
Thanks, merged to master/1.10

**Pull request closed without merging**